ordered_product/views.py

```diff
@@ -0,0 +1,98 @@
+from .models import OrderedProduct
+from client.models import Client
+from supplier.models import Supplier
+from supplier_product.models import SupplierProduct
+from delivery_location.models import DeliveryLocation
+from product.models import Product
+from django.shortcuts import redirect
+from django.shortcuts import render
+
+
+def order_list(request):
+    if isinstance(request.user, Supplier):
+        return redirect('Main Page')
+    message = ""
+    ordered_products = OrderedProduct.filter_user_name(Client(request.user))
+    if not ordered_products:
+        message = "No orders found"
+    return render(request, 'ordered_product/orders.html', {'orders': ordered_products,
+                                                           'message': message})
+
+
+def change_quantity(request, order_id):
+    try:
+        quantity = int(request.GET['quantity'])
+        ordered_product = OrderedProduct.objects.get(id=int(order_id))
+    except Exception:
+        return redirect('orders')
+    if (quantity > 0):
+        ordered_product.increase_quantity(quantity)
+    elif (quantity < 0):
+        ordered_product.decrease_quantity(abs(quantity))
+    return redirect('orders')
+
+
+def delete_order(request, order_id):
+    try:
+        OrderedProduct.objects.get(id=int(order_id)).delete_ordered_product()
+    except Exception:
```

Comment:
1. what exception do we expect here? and why?
2. there is no need for the except if the result will be the same as not having error handling. instead, use `try-finally`

Reply:
I'm not sure why I need to use finally.
I changed the code to this:
`def delete_order(request, order_id):
    try:
        OrderedProduct.objects.get(id=int(order_id)).delete_ordered_product()
    except ObjectDoesNotExist:
        pass
    return redirect('orders')`